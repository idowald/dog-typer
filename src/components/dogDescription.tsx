import * as React from "react";
import {
  Card,
  CardContent,
  createStyles,
  Theme,
  Typography,
  useMediaQuery,
  useTheme
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

// props from parent element
interface OwnProps {
  dogType?: string;
  errorMessage: string;
}
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    dogDescription: {
      width: "560px"
    },
    dogDescriptionSmall: {
      width: "calc(640px *0.5)"
    }
  })
);
const generateText = ({ dogType, errorMessage }: OwnProps): string => {
  if (errorMessage) {
    return errorMessage;
  }
  if (dogType) {
    return `Your dog type is: ${dogType}`;
  }
  return "You don't have a dog yet I believe";
};
export const DogDescription = ({ dogType, errorMessage }: OwnProps) => {
  const classes = useStyles();
  const theme = useTheme();
  const isSmallScreen = !useMediaQuery(theme.breakpoints.up("sm"));
  return (
    <Card
      className={
        isSmallScreen ? classes.dogDescriptionSmall : classes.dogDescription
      }
      variant="outlined"
    >
      <CardContent>
        <Typography variant="h5" component="h2">
          {generateText({ dogType, errorMessage })}
        </Typography>
      </CardContent>
    </Card>
  );
};
