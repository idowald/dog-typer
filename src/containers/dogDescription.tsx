import { connect } from "react-redux";
import { State } from "../store/reducers/reducer";
import { DogDescription } from "../components/dogDescription";

export const DogDescriptionContainer = connect(
  ({ dog: { dogType, errorMessage } }: State) => ({ dogType, errorMessage })
)(DogDescription);
