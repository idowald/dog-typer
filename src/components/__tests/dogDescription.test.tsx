import React from "react";
import renderer from "react-test-renderer";
import { DogDescription } from "../dogDescription";

test("renders learn react link", () => {
  const rendered = renderer.create(<DogDescription dogType={"testy"} />);
  expect(rendered.toJSON()).toMatchSnapshot();
});
