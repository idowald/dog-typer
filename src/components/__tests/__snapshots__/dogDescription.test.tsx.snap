// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-dogDescriptionSmall-2 MuiPaper-outlined MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5"
    >
      Your dog type is: testy
    </h2>
  </div>
</div>
`;
